docs/architecture/adr-007-specialization-groups.md

```diff
@@ -0,0 +1,161 @@
+beneficial to a blockchain community include: code auditing, emergency response,
+code development etc. This type of community organization paves the way for
+individual stakeholders to delegate votes by issue type, if in the future
+governance proposals include a field for issue type. 
+
+
+## Decision
+
+A specialization group can be broadly broken down into
+the following functions: 
+ - Membership Admittance
+   - via governance
+   - via special appointment
+ - Membership Acceptance
+ - Membership Revocation
+   - (probably) Without Penalty 
+     - member steps down (self-Revocation)
+     - replaced by new member from governance
+   - (probably) With Penalty 
+     - due to breach of soft-agreement (determined through governance)
+     - due to breach of hard-agreement (determined by code) 
+ - Execution of Duties
+   - Special transactions which only execute for members of a specialization group
+ - Compensation
+   - Group compensation (further distribution decided by the specialization group) 
+   - Individual compensation for all constituents of a group from the
+     greater community
+
+Election of the members of a specialization group can happen in a wide variety
+of ways and be subject to an arbitrary number of associated rules. For its
+initial implementation as a part of this ADR we recommend that a general
+election abstraction (`Electionator`) is provided as well as a basic
+implementation of that abstraction which allows for a continuous election of
+members of a specialization group. 
+
+``` golang
+// The Electionator abstraction covers the concept space for 
+// a wide variety of election kinds.  
+type Electionator interface {
+    
+    // is the election object accepting votes.
+    Active() bool 
+
+    // functionality to execute for when a vote is cast in this election, here
+    // the vote field is anticipated to be marshalled into a vote type used 
+    // by an election. 
+    Vote(addr sdk.AccAddress, vote []byte) 
+
+    // here lies all functionality to authenticate and execute changes for
+    // when a member accepts being elected
+    Accept(sdk.AccAddress) 
+
+    // Register a revoker object
+    RegisterRevoker(Revoker)
+
+    // No more revokers may be registered after this function is called
+    SealRevokers()
+
+    // register hooks to call when an election actions occur
+    RegisterHooks(ElectionatorHooks) 
+
+    // query for the current winner(s) of this election based on arbitrary
+    // election ruleset
+    QueryWinners() []sdk.AccAddress 
+
+    // query metadata for an address in the election this 
+    // could include for example position that an address
+    // is being elected for within a group   
+    // 
+    // this metadata may be directly related to 
+    // voting information and/or privileges enabled
+    // to members within a group. 
+    QueryMetadata(sdk.AccAddress) []byte
+}
+
+type ElectionatorHooks interface {
+    VoteCast(addr sdk.AccAddress, vote []byte)
```

Comment:
these 3 hooks are missing the proposal ID. Otherwise, you won't be able to distinguish similar events from one proposal to another

Reply:
So there isn't the concept of proposalIDs here - these vote belong _exclusively_ to the Electionator (which itself can be thought of as a single proposal)

Reply:
I don't follow. Can there be multiple proposals to register more members?

Reply:
There are no proposals here. Just votes which pertain specifically to one electionator. Anyone can create and send a vote to the electionator item which will presumably attempt to marshal those bytes into a particular struct and apply the vote information in some arbitrary way. There can be multiple Electionators within the Cosmos-Hub for multiple specialization groups, votes would need to be routed to the Electionator _upstream_ of here.